modules/calendar/html_parser.py

```diff
@@ -0,0 +1,31 @@
+import re
+from typing import Generator
+
+__MD_LINK_REGEX = re.compile(
+	# Group 1: The label
+	# Group 2: The full URL
+	r"\[(.*?[^\\])\]\((https?:\/\/.*?)\)"
+)
+__HTML_LINK_REGEX = re.compile(
+	# Group 2|5: The full URL
+	# Group 3|6: A truncated URL, if the label was the same as the URL
+	# Group 4: The label if it was not the same as the URL
+	r"(?:<a\s+(?:[^>]*?\s+)?href=(['\"])(.*?)\1.*?>(?:(?=\2)https?://(.*?(?=[/<#?])[^<?#]{0,10}).*?|(.*?))<\/a>|(https?://([\w.@:]+[^\s()\[\],!<>|\"'{}]{0,9})[^\s()\[\],!<>|\"'{}]*))"
+)
+
+
+def md_links_to_html(text: str) -> str:
+	return __MD_LINK_REGEX.sub(r"<a href='\2'>\1</a>", text)
+
+
+def html_links_to_md(text: str) -> str:
+	return __HTML_LINK_REGEX.sub(r"[\3\4\6](\2\5)", text)
```

Comment:
Title text was removed making it so partial links are shown when hovering instead of the full link.

Having something like `[\3\4\6](\2\5 "\2\5")` would show the full link on hover.

Also, is there a reason to get rid of the `...` after the shortened links?

Reply:
I changed it to include these in #151, feel free to leave a review if you intentionally changed it

Reply:
Reason I skipped ... Was because I couldn't figure out how I could do it, didn't know u could pass a function into sub